boa_engine/src/object/shape/shared_shape/forward_transition.rs

```diff
@@ -6,7 +8,28 @@ use crate::object::JsPrototype;
 use super::{Inner as SharedShapeInner, TransitionKey};
 
 /// Maps transition key type to a [`SharedShapeInner`] transition.
-type TransitionMap<T> = FxHashMap<T, WeakGc<SharedShapeInner>>;
+#[derive(Debug, Trace, Finalize)]
+struct TransitionMap<T: Debug + Trace + Finalize> {
+    map: FxHashMap<T, WeakGc<SharedShapeInner>>,
+    data: u8,
+}
+
+impl<T: Debug + Trace + Finalize> Default for TransitionMap<T> {
+    fn default() -> Self {
+        Self {
+            map: FxHashMap::default(),
+            data: 0,
+        }
+    }
+}
+
+impl<T: Debug + Trace + Finalize> TransitionMap<T> {
+    fn get_and_increment_count(&mut self) -> u8 {
+        let result = self.data;
+        self.data = self.data.wrapping_add(1);
+        result
+    }
+}
```

Comment:
If we prune on `map.len() > 255`, this will prune every time after 255 insertions. But with having the data field (probably should name it to `counter`) it will prune on every 255 insertions, since it intentionally overflows to `0`.